### ISSUE TITLE
tcl syntax highlighting is buggy

### ISSUE THREAD

**Issue by Akuli** [Akuli/porcupine#310]
https://user-images.githubusercontent.com/18505570/107215213-b3e53d00-6a13-11eb-9223-9a7a90d8f944.mp4

**Comment by Akuli** [comment]
happened again

https://user-images.githubusercontent.com/18505570/108481616-16a2c800-72a1-11eb-9af7-30322722f42c.mp4

**Comment by Akuli** [comment]
Can't reproduce now. I think I just hadn't pulled in a while:

```
*   289a788 Merge remote-tracking branch 'origin/master'
|\  
| * b301dad fix Highlight bugs (#364)
* | 00d83f9 ignore embedded windows when counting chars (fixes #354)
|/  
* b7f7b69 Tab closing right-click menu (#365)
```

Note that 00d83f9 is the commit I referred to above. I improved the tests in 0f1c1d7 to be sure.
